src/Sulu/Bundle/AdminBundle/Resources/js/services/Router/Router.js

```diff
@@ -132,6 +132,8 @@ export default class Router {
     redirectFlag: boolean = false;
 
     constructor(history: Object) {
+        makeObservable(this); // this should be enough that we still should be able to use legacy decorators
+
```

Comment:
this should make the above variables observable, but the beneath test fails `expect(isObservable(router.route)).toBe(true)` as router.route is still not observable based on isObservable.

Reply:
Looks like this error maybe related to some babel issues as I tried now define the observable via makeObservable directly and run into the following error:

<img width="812" alt="Bildschirmfoto 2023-11-29 um 01 00 40" src="https://github.com/sulu/sulu/assets/1698337/4c17b42c-bf3e-4741-8008-1ddb36791706">

Reply:
Strange is that it still output the route is defined:

but as expected it is undefined:

<img width="470" alt="Bildschirmfoto 2023-11-29 um 01 19 15" src="https://github.com/sulu/sulu/assets/1698337/7966c52a-aa82-4450-b78e-d69803acca5a">

Reply:
there are some subtleties in how Babel defines its class fields. From top of my head in 'loose' mode, it only allocates the field on first assignment, so you might want to set this to 'strict' (or the other way around, it has been a while). Explicitly initializing it might also help, e.g. `route: ?Route = undefined`.

Reply:
Also you probably want to check `expect(isObservableProp(router, "route"))`, otherwise it checks if the value stored in the route field is observable, not whether the property itself is.

Reply:
Thx @mweststrate for your response. It really seems to be that in this case `isObservable` is failing but your suggested `expect(isObservableProp(router, "route"))` works. So in Mobx 6 (or 5) only the class itself knows if its observable property not the property itself?

It look like / I hope it is only used in tests and so I can maybe just do a replace via regex to convert existing `isObservable` to `isObservableProp`:

```
\(isObservable\((.*)\.(.*)\)\)\.
(isObservableProp($1, '$2')).
```

<img width="609" alt="Bildschirmfoto 2023-11-30 um 10 24 06" src="https://github.com/sulu/sulu/assets/1698337/2f56ddb3-bb97-4a3b-b5ae-a4c1082dcbed">

Following changes seems to make the `Router.test.js` work: https://github.com/sulu/sulu/pull/7211/commits/b8cd4860eb2618dff5279da5abbf2ee98238d2d8. Seems I need to check this on a lot of other tests.


If props and params are never itself observable is the replacement of `isArrayLike` not required:

```diff
-if (!isArrayLike(values)) {
+if (!(Array.isArray(values) || isObservableArray(values))) {
```

and could it just be:

```diff
+if (!(Array.isArray(values)) {
```